main.cpp

```diff
@@ -24,46 +26,54 @@
 static llvm::LLVMContext context;
 
 int main() {
-	std::cout << "Hello world" << std::endl;
-	llvm::SMDiagnostic err;
-	auto file =  llvm::StringRef("/home/alex4o/bitcode_experiments_cxx/file.bc");
-	std::cout << file.str() << std::endl;
-	std::unique_ptr<llvm::Module> m = llvm::parseIRFile(file, err, context);
-	llvm::Module result(llvm::StringRef("output"), context);
-	
-	if(m)
-	{
-		m->print(llvm::outs(), nullptr);
-		for(auto &function: m->functions()) {
-			std::cout << function.getName().str() << std::endl;
-			if(function.getName().size() < 2) {
-	
-				auto resultCallee = result.getOrInsertFunction(function.getName(), function.getReturnType());
-				llvm::Function* resultFunction = llvm::dyn_cast_or_null<llvm::Function>(resultCallee.getCallee());
+  llvm::SMDiagnostic err;
+  auto file = llvm::StringRef("/home/alex4o/bitcode_experiments_cxx/file.bc");
+  std::cout << file.str() << std::endl;
+  std::unique_ptr<llvm::Module> um = llvm::parseIRFile(file, err, context);
+  const llvm::Module *m = um.get();
+
+  llvm::Module result(llvm::StringRef("output"), context);
+
+  result.setDataLayout(m->getDataLayout());
+  result.setTargetTriple(m->getTargetTriple());
+  result.setModuleInlineAsm(m->getModuleInlineAsm());
 
-				llvm::ValueToValueMapTy VMap;
-				auto resultArgs =  resultFunction->arg_begin();
-				for(llvm::Value &arg : function.args()) {
-					VMap[&*resultArgs++] = &arg;;;
-				}
-				
-				llvm::SmallVector<llvm::ReturnInst*, 8> returns;
+  llvm::ValueToValueMapTy VMap;
+  if (m) {
+    for (const llvm::Function &function : *m) {
+      if(function.getName().size() < 2) {
+        std::cout << function.getName().str() << std::endl;
 
-				llvm::CloneFunctionInto(resultFunction, &function, VMap, false, returns);
+        auto resultCallee = result.getOrInsertFunction(
+            function.getName(), function.getReturnType());
+        llvm::Function *resultFunction =
+            llvm::dyn_cast_or_null<llvm::Function>(resultCallee.getCallee());
+        if(resultFunction == nullptr) {
+          std::cout << "ERROR" << std::endl;
```

Comment:
Should we be failing on the first error rather than burying it?

Reply:
That code got removed in the later commits, as it was rewritten in a way that couldn't fail.